Added the possibility to configure the download-link and version for Plugins-Manager and Command-Runner

malice00 opened the pull request:
This PR adds the possibility to configure the download-link and version for the JMeter Tools 'Plugins-Manager' and 'Command-Runner', similar to how JMeter itself can be configured.
This is useful if you are running Taurus on a computer that is not allowed to have direct access to the internet and needs to get all files from an internal mirror.

Since I don't have much Python experience, I couldn't figure out how to write tests for this change.

Quick checklist:
- [X] Description of PR explains the context of change
- [X] Unit tests cover the change, no broken tests
- [ ] No static analysis warnings (Codacy etc.)
- [X] Documentation update
- [X] Changes file inside `site/dat/docs/changes` directory, one-line note of change inside

greyfenrir commented:
Hello. Great, for people who haven't much python experience it looks very clean)
Only two notes:
1. documentation. as you mentioned `plugins-manager` and `command-runner` in the config example would be great to explain what they are to end user. At the moment it looks little bit hacky, you know.
2. unit tests. I agree, it isn't obvious. Please use [this file](https://github.com/Blazemeter/taurus/blob/master/tests/unit/modules/jmeter/test_JMeterTool.py) as example. The following must be checked: changing of plugins-manager and command-runner download links/versions causes changing of the way of getting them. I guess something should be added to in MockJMeter to implement the check. If it will be difficult for you don't hesitate to ask me for help.

malice00 commented:
Added some tests -- I think I figured it out, please let me know if they are good (enough).
Added a couple of lines of documentation about the 2 tools as well.

Updated checklist:
- [X] Description of PR explains the context of change
- [X] Unit tests cover the change, no broken tests
- [ ] No static analysis warnings (Codacy etc.)
- [X] Documentation update
- [X] Changes file inside `site/dat/docs/changes` directory, one-line note of change inside

Still not sure how to do static analysis, is this not included in the build process?

greyfenrir commented:
Hi. Sorry for waiting, I gonna merge it. About static analysis: you're right, this isn't included unfortunately.
